### PR TITLE
DashPay: Profile Picture | Select Picture

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

NMA-418

Show a menu that allows the user to take a picture or to choose an existing picture.  The path of the image is saved to the shared preferences.

**Known issues:**
- The top of the selection dialog does not have rounded corners

## Related PR's and Dependencies

None

## Screenshots / Videos

Screenshot_1602598936a

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/Configuration.java [hunk 1]
```diff
@@ -431,7 +432,15 @@ public long getLastSeenNotificationTime() {
         return prefs.getLong(PREFS_LAST_SEEN_NOTIFICATION_TIME, 0);
     }
 
-    public void setPrefsLastSeenNotificationTime(long lastSeenNotificationTime) {
+    public void setLastSeenNotificationTime(long lastSeenNotificationTime) {
         prefs.edit().putLong(PREFS_LAST_SEEN_NOTIFICATION_TIME, lastSeenNotificationTime).apply();
     }
+
+    public void setLocalProfilePictureUri(String picturePath) {
+        prefs.edit().putString(PREFS_KEY_LOCAL_PROFILE_PICTURE_URI, picturePath);
```
Comment (sambarboza) [inline common/src/main/java/org/dash/wallet/common/Configuration.java:440]: `.apply()` needs to be called to save the image path to preferences.

Comment (HashEngineering) [reply common/src/main/java/org/dash/wallet/common/Configuration.java]: Fixed here: https://github.com/dashevo/dash-wallet/pull/528/commits/649f0091a5dc9d8caaf4046096c9d15993998ff3

File: wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt [hunk 2]
```diff
@@ -177,4 +221,115 @@ class EditProfileActivity : BaseMenuActivity() {
...
+                when {
+                    grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED -> choosePicture()
+                    else -> choosePictureWithPermission()
+                }
+            }
+            else -> super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        }
+    }
+
+    private fun gerFileUri(file: File): Uri {
```
Comment (sambarboza) [inline wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt:332]: I think the `ger...` is a typo?

Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt]: Fixed here:  https://github.com/dashevo/dash-wallet/pull/528/commits/206c8d1fd6aee934a93541071dbfbd893a4c6004

**Review by sambarboza — APPROVED** [review]

**Review by tomasz-ludek — APPROVED** [review]

**Pull request merged**